### ISSUE TITLE
new mongoose.Schema() is throwing: Cannot read property 'options' of undefined at...

### ISSUE THREAD

**Issue by Nektoram** [Automattic/mongoose#7170]
Hello everyone,

I want to report a bug that randomly occured today, it worked completely fine yesterday and the only thing I did with the project was move the folder it was inside.

I am using Mongoose inside of Electron.js for a desktop application and what happens when I try to run this app is this:

![cef40660567fccc4a22c69380ea8787d](https://user-images.githubusercontent.com/39199931/47325513-6d158280-d664-11e8-87c6-ff3a6e88261a.png)

The app fails loading up and I get the error above. I am very, very sure that the source of the problem is the mongoose.Schema() method, since I don't get this error and the app loads up fine when I remove every call to mongoose.Schema() inside of my project. Obviously this can't be the solution since my whole app is based on MongoDB with mongoose and removing every Schema would also mean that I would probably have to completely switch over to another Database Service or somehow recode everything which I wouldn't want as of now. Here are my two schemas:

![3c9119a74082f85532ee22a896c9c617](https://user-images.githubusercontent.com/39199931/47325652-33914700-d665-11e8-8da1-696f413b624a.png)

![4f7c90b20278610a7a552e50811460a8](https://user-images.githubusercontent.com/39199931/47325733-94208400-d665-11e8-8c2c-291f098334e9.png)

and like I said, the error only goes away when I remove both of the calls to mongoose.Schema() in both files. If I did anything wrong on my end with configuring mongoose.Schema correctly which is leading to that error, I apologize and would be thankful for improvements. Otherwise, any ideas on how to fix this? I am not sure about that but it seems to have stopped working since I have moved the project folder onto my second drive. It shouldn't be the node_modules though, since I deleted them, reinstalled them via npm install and checked that especially mongoose was fully up to date. I hope that this can be fixed as soon as possible since I really need to keep working on this project.

Thanks in advance! 👍

**Comment by Nektoram** [comment]
Alright, seems like I've fixed this issue myself now. If anyone has the same issue and is interested in how I fixed it: I just changed the problematic line of code myself, it's inside of this file: node_modules/mongoose/lib/schema.js - Line 332 was the one that caused the issue. Changed it from it's original: `strict: 'strict' in this.base.options ? this.base.options.strict : true,` which caused the error shown in the first screenshot above to:  `strict: this.base ? this.base.options.strict : false,` now and everything is working perfectly fine. Closing this issue since it appears to be fixed like this, although I am REALLY perplexed of why this happened to me in the first place.
